Issue 8.1

adnahapet opened the pull request:
createOrg update

adnahapet committed:
initialized join pugs/routers/controllers

adnahapet committed:
Updated org controller

adnahapet committed:
pug updates

adnahapet committed:
Split data sent for organization

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -0,0 +1,56 @@
+      var user = {
+          "email": req.body.email,
+      }
+      .post('localhost:5000/v1/admin/organizations')
```
Change route to be 'aws_apiserver_ip:5000/organizations'

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -0,0 +1,56 @@
+          "email": req.body.email,
+      }
+      .post('localhost:5000/v1/admin/organizations')
+      .send(organization,user)
```
send the data in this format:
```
{organization: organization, user: user}
```

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -0,0 +1,56 @@
+                if (err) {
+                    console.log(err);
+                }
+                else if(doc){
```
Try to branch on statusCode, so something like if (res.statusCode == 400) {} or try printing the resp.body when trying to create an org with an org name that already exists and see what it looks like, I am unsure as blueprint creates the response.

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -0,0 +1,56 @@
+                    console.log("Sorry, Organization Already Exists!");
+                }
+                else {
+                    result = resp.body.result;
```
The successful response body looks as follows:
```
{
  organization: {
    org_id: 'id',
    admin_id: 'id'
  }
}
```
So you can access those values via, req.body.organization.org_id, etc.

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -0,0 +1,56 @@
+                    console.log("Organization Successfully Registered");
+                }
+
+                if(result){
```
Make sure the render statements work with the new changes you make.

DannyPeck commented on app/routers/TestRouter.js:
```diff
@@ -0,0 +1,11 @@
+var passport = require ('passport')
```
We can remove this, unless it serves some purpose I am unaware of.

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -0,0 +1,56 @@
+
+  return function (req, res) {
+
+      var organization = {
```
I am assuming these grab the correct values :)

DannyPeck reviewed (changes requested):
I think we are really close, just refactor a few things

adnahapet committed:
Fixed CreateOrg

adnahapet committed:
Merge branch 'issue17' into issue-8.1

DannyPeck commented on app/controllers/JoinController.js:
```diff
@@ -25,27 +25,19 @@ JoinController.prototype.join = function () {
+        .send({organization: organization,user: user})
+        .end(function (err, resp) {
 
                 if (err) {
```
This section is too tabbed in ;)

adnahapet committed:
Fixed err handling

Changed indentation and added redirect to (err) case

adnahapet committed:
Merge branch 'issue20' into issue-8.1

adnahapet committed:
Nuked unused pugs

merged form from create user into dashboard

adnahapet committed:
Error handling on create user

changed res.render(s)

adnahapet committed:
pug changes

removed deprecated get

DannyPeck reviewed (approved):
Everything looks great!

adnahapet committed:
Appearance update

added local asset folder and added some brand-appropriate style
elements to the login.pug
